### PR TITLE
Mbasiuk/tdi 39295 dev

### PR DESCRIPTION
**Pull request by mbasiuk-talend**
**What is the current behavior?** (You can also link to an open issue here)
Issue's:
https://jira.talendforge.org/browse/TDI-39687
https://jira.talendforge.org/browse/TDI-39371
https://jira.talendforge.org/browse/TDI-39523
https://jira.talendforge.org/browse/TDI-39407
https://jira.talendforge.org/browse/TDI-39717

**What is the new behavior?**
Fixes for found issues

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1437/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1437/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1565/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1565/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1574/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1574/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowReader.java [hunk 1]
```diff
@@ -77,28 +110,49 @@ public boolean start() throws IOException {
...
+        if (!validationResult) {
+            return false;
+        }
+        boolean hasNext = false;
         try {
             hasNext = rs.next();
         } catch (SQLException e) {
-            throw new IOException(e);
+            if (dieOnError) {
+                throw new IOException();
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowReader.java]: empty exception message here.
Could you send parent exception or its message to constructor?

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowWriter.java [hunk 2]
```diff
@@ -173,54 +171,41 @@ public void write(Object object) throws IOException {
 
     private void handleSuccess(IndexedRecord input) throws SQLException {
 
-        Schema outputSchema = rowProperties.getSchema();
-        if (outputSchema == null || outputSchema.getFields().size() == 0) {
+        if (mainSchema == null || mainSchema.getFields().size() == 0) {
             return;
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowWriter.java]: Not sure we need check schema in every iteration

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowWriter.java [hunk 3]
```diff
@@ -300,14 +284,11 @@ public Result close() throws IOException {
...
         try {
 
-            cleanWrites();
-
-            if (commitCount > 0 && connection != null && statement != null) {
-                if (!rowProperties.propagateQueryResultSet()) {
-                    statement.executeBatch();
-                }
+            if (commitStep > 1 && commitCounter > 0 && connection != null && statement != null) {
+                cleanWrites();
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRowWriter.java]: Please clean it in every case

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1586/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1586/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1587/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1587/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1588/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1588/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Ambasiuk%2FTDI-39295_dev)

**Pull request merged**